test/e2e-tests.sh

```diff
@@ -85,6 +85,38 @@ $(dirname $0)/upload-test-images.sh e2e || fail_test "Error uploading test image
 # Setup resources common to all eventing tests
 setup_events_test_resources|| fail_test "Error setting up test resources"
 
-go_test_e2e ./test/e2e || fail_test
+go_test_e2e ./test/e2e
+exit_result=$?
+if [ ${exit_result} -ne 0 ]; then
+  # Collecting logs from all knative's eventing pods
+  echo "====== Collecting logs for namespace: knative-eventing ========"
+  for pod in $(kubectl get pod -n knative-eventing | grep Running | awk '{print $1}' ); do
```

Comment:
Do we somewhere dump the state of the pods that are not running? Say something goes wrong, could / should we do something like describe on those pods?

Reply:
that seems like a good feature to add later, I will add an issue.